File: KeySerializationSchema/ComponentTest/Models/OutputEvent.cs
using Newtonsoft.Json;

namespace KeySerializationSchema.ComponentTest.Models;

public class OutputEvent
{
    public OutputEvent(string deviceId, double value, long timestamp, string keys, string extraField)
    {
        DeviceId = deviceId;
        Value = value;
        Timestamp = timestamp;
        Keys = keys;
        ExtraField = extraField;
    }

    [JsonProperty("deviceId")]
    public string DeviceId { get; set; }

    [JsonProperty("value")]
    public double Value { get; set; }
    
    [JsonProperty("timestamp")]
    public long Timestamp { get; set; }
    
    [JsonProperty("keys")]
    public string Keys { get; set; }

    [JsonProperty("extraField")]
    public string ExtraField { get; set; }
    

}

File: SerializationIssue/src/main/java/com/cheetah/example/model/OutputEvent.java
package com.cheetah.example.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.ArrayList;
import java.util.List;

/** OutputEvent represents the events being generated. */
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class OutputEvent {
    public OutputEvent(InputEvent inputEvent, String extraFieldValue){
        this.deviceId = inputEvent.getDeviceId();
        this.value = inputEvent.getValue();
        this.timestamp = inputEvent.getTimestamp();
        this.list = inputEvent.getList();
        this.extraField = extraFieldValue;
    }
    private String deviceId;
    private double value;
    private long timestamp;
    private List<String> list;
    private String extraField;
}
